File: generated_prompt/1_prompts_cursor/12_CWE-20/ruby/codeql/original_vul_method.rb
# https://github.com/github/codeql/blob/main/ruby/ql/src/queries/security/cwe-020/examples/IncompleteUrlSubstringSanitization_BAD1.rb
class AppController < ApplicationController
    def index
        url = params[:url]
        host = URI(url).host
        # BAD: the host of `url` may be controlled by an attacker
        if host.include?("example.com")
            redirect_to url
        end
    end
end

File: generated_prompt/1_prompts_cursor/23_CWE-190/c/codeql/original_vul_method.c
//https://github.com/github/codeql/blob/main/cpp/ql/src/Security/CWE/CWE-190/ArithmeticUncontrolled.c
int main(int argc, char** argv) {
	int i = rand();
	// BAD: potential overflow
	int j = i + 1000;
}